### PR TITLE
Get activations

### PR DESCRIPTION
**Pull request by kvetab**
After discussing with David (@prihoda), I'm creating a PR based on the issue #153 .

Enable returning files with activations for secondary structure and disorder predictions.
Config file can contain parameter `get_activations`, which controls whether these files are created.
Use softmax on raw activations tensors (`yhat_...`) to obtain "probabilities" for each of the classes. Return these from the `get_secondary_structure` function in `BasicAnnotationExtractor`, store them in `BasicSecondaryStructureResult` and subsequently in `BasicExtractedAnnotations`, which is returned by `get_annotations`.
In `predict_annotations_using_basic_models`, check for `get_activations` parameter; if true, create output files for activations from DSSP3, DSSP8 and disorder. Then, store activations into dataframes along with the sequence ID and residue number. Each row corresponds to one position in one sequence, columns correspond to sequence ID, residue number (1-based) and all the possible classes.
For each prediction type (DSSP3, DSSP8 and disorder), concatenate all dataframes and print to the corresponding csv file.

Looking forward to any comments, especially on the use of the config file, I'm a bit fuzzy on that.

### CODE REVIEW

**Comment by sacdallago** [comment]
Thanks @kvetab . Sorry for the delay, somehow this didn't show up on my notifications 😮 I'll put it on the todo for this week :)

File: bio_embeddings/extract/pipeline.py [hunk 1]
```diff
@@ -415,14 +439,29 @@ def predict_annotations_using_basic_models(model: str, **kwargs) -> Dict[str, An
             DSSP3_sequence = deepcopy(protein_sequence)
             DSSP3_sequence.seq = Seq(convert_list_of_enum_to_string(annotations.DSSP3))
             DSSP3_sequences.append(DSSP3_sequence)
+            DSSP3_raw_df = DataFrame(annotations.DSSP3_raw[:, :, 0].detach().numpy().transpose(),
```
Comment (konstin) [inline bio_embeddings/extract/pipeline.py]: This needs `.cpu()` after detach to work with the GPU (also the two below)

File: bio_embeddings/extract/pipeline.py [hunk 2]
```diff
@@ -434,6 +473,12 @@ def predict_annotations_using_basic_models(model: str, **kwargs) -> Dict[str, An
     write_fasta_file(DSSP8_sequences, DSSP8_predictions_file_path)
     write_fasta_file(disorder_sequences, disorder_predictions_file_path)
 
+    if 'get_activations' in kwargs and kwargs['get_activations']:
+        # create files with activations for each multiclass prediction
+        concatenate_dataframe(DSSP3_raw).to_csv(DSSP3_raw_predictions_file_path, index=False)
+        concatenate_dataframe(DSSP8_raw).to_csv(DSSP8_raw_predictions_file_path, index=False)
+        concatenate_dataframe(disorder_raw).to_csv(disorder_raw_predictions_file_path, index=False)
```
Comment (konstin) [inline bio_embeddings/extract/pipeline.py]: We generally use the id as index, this should write the same format as we use elsewhere:

```suggestion
        concatenate_dataframe(DSSP3_raw).set_index('seqID').rename_axis(None).to_csv(DSSP3_raw_predictions_file_path)
        concatenate_dataframe(DSSP8_raw).set_index('seqID').rename_axis(None).to_csv(DSSP8_raw_predictions_file_path)
        concatenate_dataframe(disorder_raw).set_index('seqID').rename_axis(None).to_csv(disorder_raw_predictions_file_path)
```

**Review by konstin — APPROVED** [review]
Thanks and so sorry it took so long, looks good with two minor changes

File: bio_embeddings/extract/pipeline.py (hunk 1 above)
Comment (kvetab) [reply bio_embeddings/extract/pipeline.py]: Fixed

File: bio_embeddings/extract/pipeline.py (hunk 2 above)
Comment (kvetab) [reply bio_embeddings/extract/pipeline.py]: Fixed

**Comment by konstin** [comment]
Thank you!

**Comment by kvetab** [comment]
Hi @sacdallago, @konstin,
I'd like to ask, how often / in which situations do you make new releases? It would be really cool if you could make one sometime soon, as I'd like to use the contents of my PR in a project without having to install from the develop branch :)
Thanks
